packages/app/components/Queue/QueueList.tsx

```diff
@@ -170,91 +199,63 @@ export default function QueueList({ role, onOpenClick }: QueueListProps) {
         <HeaderRow justify="space-between">
           <QueueTitle>Helping</QueueTitle>
           <div>
-            <AlertButton danger size="large">
+            <AlertButton danger size="large" onClick={alertHelpingAll}>
               Alert All
             </AlertButton>
-            <Button type="primary" size="large">
+            <Button type="primary" size="large" onClick={finishHelpingAll}>
               Finish All
             </Button>
           </div>
         </HeaderRow>
-        <StudentInfoCard />
-        <GroupQuestions />
+        {currentQuestion && (
+          <div>
+            <StudentInfoCard
+              updateQuestion={updateQuestionTA}
+              alertStudent={alertStudent}
+              question={currentQuestion}
+            />
+            <GroupQuestions />{" "}
+          </div>
+        )}
       </Col>
     );
   };
 
-  return (
-    <div>
-      <Row gutter={[64, 64]}>
-        <Col flex="auto" order={screens.lg === false ? 2 : 1}>
-          <Row justify="space-between">
-            <QueueTitle>Queue 1</QueueTitle>
-            {role === "student" && (
-              <Link href="/queue/join">
-                <Button type="primary" size="large">
+  return useMemo(() => {
+    return (
+      <div>
+        <Row gutter={[64, 64]}>
+          <Col flex="auto" order={screens.lg === false ? 2 : 1}>
+            <Row justify="space-between">
+              <QueueTitle>Queue 1</QueueTitle>
+              {role === "student" && (
+                <Button type="primary" size="large" onClick={joinQueue}>
                   Join Queue
                 </Button>
-              </Link>
-            )}
-          </Row>
-          {role === "ta" && !helping && renderTAHeader()}
-          {role === "ta" && helping && renderHelpingHeader()}
-          {role === "student" && renderStudentHeader()}
+              )}
+            </Row>
+            {role === Role.TA && !helping && renderTAHeader()}
+            {role === Role.TA && helping && renderHelpingHeader()}
+            {role === Role.STUDENT && renderStudentHeader()}
```

Comment:
This totally works but I'm curious if it would be easier long term to have a single if statement on the `role` attribute that could render both the TA Header and the QueueCard in the same file instead of separating the logic into different files? This would mean making a separate card for the student and ta (`StudentQueueCard` and `TAQueueCard`) so I'm not sure it's worth the refactoring effort at the moment but just thought I'd throw it out there.

Reply:
I'm not sure if I completely understand, but the headers and queue card rendering logic has to be separate because there's only one header for a list of queue cards. I found that there were a lot of similarities between the StudentQueueList/StudentQueueCard and the TAQueueList/TAQueueCard so I abstracted them into the same files. I'm not sure if this detracts from readability though, lmk if you have any further thoughts on this.

Reply:
That makes sense. Looking back to my original comment I wasn't so clear. 
What I was thinking was that it could more readable to take each function in the `QueueCard` like `renderTACard()`, `renderHelpingCard()`, `renderStudentCard()` and separate them into their own components. Then in the `QueueList` you could have something like 
```
{if (role === Role.TA && !helping) {
  renderTAHeader();
  {questions.map((question: Question, index: number) => {
     return (<TAQueueCard />)   // which would be the same as the renderTACard() function
  }
}
```
It would add some more code to the `QueueList` component so I'm not sure it would be that much better. I'm curious to hear what people think.

Reply:
Oh I see! I think that's valid and might be more readable. I'm inclined to keep the `QueueCards` in the same component for now though so that we can reuse the same styled-components. I also think it might not be worth the refactoring effort at the moment, but I'd be happy to revisit this if it poses issues later on.